### PR TITLE
fix(curriculum): rework Project Euler 94

### PR DESCRIPTION
**Pull request by gikf**
- Reworks challenge to use argument in function call.
- Adds more tests.
- Adds solution.
- Related to #40896.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

**Review by marsonya — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-94-almost-equilateral-triangles.md [hunk 1]
```diff
@@ -12,37 +12,73 @@ It is easily proved that no equilateral triangle exists with integral length sid
 
 We shall define an <dfn>almost equilateral triangle</dfn> to be a triangle for which two sides are equal and the third differs by no more than one unit.
 
-Find the sum of the perimeters of all <dfn>almost equilateral triangle</dfn> with integral side lengths and area and whose perimeters do not exceed one billion (1,000,000,000).
+Find the sum of the perimeters of all <dfn>almost equilateral triangle</dfn> with integral side lengths and area and whose perimeters do not exceed `limit`.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-94-almost-equilateral-triangles.md:15]:
```suggestion
Find the sum of the perimeters of all <dfn>almost equilateral triangles</dfn> with integral side lengths and area and whose perimeters do not exceed `limit`.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-94-almost-equilateral-triangles.md [hunk 2]
```diff
@@ -12,37 +12,73 @@ It is easily proved that no equilateral triangle exists with integral length sid
 
 We shall define an <dfn>almost equilateral triangle</dfn> to be a triangle for which two sides are equal and the third differs by no more than one unit.
 
-Find the sum of the perimeters of all <dfn>almost equilateral triangle</dfn> with integral side lengths and area and whose perimeters do not exceed one billion (1,000,000,000).
+Find the sum of the perimeters of all <dfn>almost equilateral triangles</dfn> with integral side lengths and area and whose perimeters do not exceed `limit`.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-94-almost-equilateral-triangles.md:15]: A quick question before I approve - do we need to use the `dfn` tag more than once?

Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-94-almost-equilateral-triangles.md:15]: Most likely no, I guess the tag in the middle (in line 13) should stay and remove it from elsewhere.

**Review by Sembauke — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Comment by moT01** [comment]
I added this solution to the forum guide article

**Pull request merged**